### PR TITLE
Token identification

### LINKED ISSUE #6: Identify details of contracts that are not ERC721
Check and give custom errors if contract addresses are
- [x] ERC20
- [ ] ERC1155
- [x] Is EOA

### LINKED ISSUE #7: Don't crash if ERC721 does not have IERC721Metadata

### PR DESCRIPTION
**Pull request by 0xjimmy**
Fixes #6 and Fixes #7.
Doesn't crash if missing metadata interfaces (Name, URI)
Now show if the entered NFT address is an ERC721, ERC20, EOA or an unidentified contract.

### CODE REVIEW

File: app/ts/components/Transfer.tsx [hunk 1]
```diff
@@ -68,9 +68,13 @@ export const Transfer = ({ provider, blockInfo }: { provider: Signal<ProviderSto
...
-				return selectedNft.value = nft
+			const identifiedAddress = await itentifyAddress(address, id, provider.value?.provider)
+			if (identifiedAddress.address === addressInput.value && identifiedAddress.inputId === idInput.value) {
+				if (identifiedAddress.type === 'ERC721') {
+					selectedNft.value = identifiedAddress
+				} else {
+					fetchingStates.value = identifiedAddress.type
+				}
 			}
 		} catch (e: any) {
```
Comment (KillariDev) [inline app/ts/components/Transfer.tsx:67]: no anys

File: app/ts/library/identifyTokens.ts [hunk 2]
```diff
@@ -1,47 +1,143 @@
...
+				address,
+				name: erc20Interface.decodeFunctionResult('name', name.returnData)[0],
+				symbol: erc20Interface.decodeFunctionResult('name', symbol.returnData)[0],
+				decimals: BigInt(erc20Interface.decodeFunctionResult('decimals', decimals.returnData)[0]),
+				totalSupply: erc20Interface.decodeFunctionResult('totalSupply', totalSupply.returnData)[0]
+			}
+		}
+
+	} catch (error) {
+		// For any reason decoding txing fails catch and return as unknown contract
```
Comment (KillariDev) [inline app/ts/library/identifyTokens.ts:136]: if we lose connection here, we map it incorrectly as contract too? These kind of catch alls are dangerous

**Review by KillariDev — CHANGES_REQUESTED** [review]
fix the catch alls and any's as I have mentioned about them in the previous PR's too

File: app/ts/components/Transfer.tsx [hunk 3]
```diff
@@ -32,30 +32,18 @@ export const Transfer = ({ provider, blockInfo }: { provider: Signal<ProviderSto
 	})
 
 	function validateAddressInput(event: JSX.TargetedEvent<HTMLInputElement>) {
-		try {
-			addressInput.value = getAddress(event.currentTarget.value.toLowerCase())
-		} catch (e) {
-			console.error(e)
-			addressInput.value = undefined
-		}
+		const value = event.currentTarget.value.toLowerCase().trim()
```
Comment (KillariDev) [inline app/ts/components/Transfer.tsx:35]: validateRecipientInput and validateAddressInput seem like the same function, could just use one that validates address

**Review by KillariDev — APPROVED** [review]

File: app/ts/components/Transfer.tsx (hunk 3 above)
Comment (0xjimmy) [reply app/ts/components/Transfer.tsx:35]: Event handler for different input box.

**Pull request merged**